### ISSUE TITLE
Stack overflow in worker service if ExecuteAsync throws an exception

### ISSUE THREAD

**Issue by TETYYS** [Barsonax/Singularity#130]
To reproduce:

Create a new worker service - `dotnet new worker`
Install `Singularity.Microsoft.DependencyInjection` (v0.17.1)
Worker.cs -
```
using System;
using System.Threading;
using System.Threading.Tasks;
using Microsoft.Extensions.Hosting;

namespace test
{
    public class Worker : BackgroundService
    {

        public Worker(IServiceProvider Provider)
        {
        }

        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
        {
            throw new Exception("a");
        }
    }
}
```

Program.cs -

```
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using Singularity;

namespace test
{
    public class Program
    {
        public static void Main(string[] args)
        {
            CreateHostBuilder(args).Build().Run();
        }

        public static IHostBuilder CreateHostBuilder(string[] args) =>
            Host.CreateDefaultBuilder(args)
                .UseServiceProviderFactory(new SingularityServiceProviderFactory(SingularitySettings.Microsoft))
                .ConfigureServices((hostContext, services) =>
                {
                    services.AddHostedService<Worker>();
                });
    }
}
```

Workaround - don't throw exceptions in `ExecuteAsync`

**Comment by Barsonax** [comment]
Strange doesn't immediately looks like its related to Singularity.

Will have to check it later.

**Comment by Barsonax** [comment]
Reproduced and probably found the fix already.

**Comment by Barsonax** [comment]
Fixed in 0.17.2, thanks for reporting @TETYYS
